webdriver/tests/bidi/script/evaluate/serialization.py

```diff
@@ -28,3 +28,96 @@ async def test_primitive_values(bidi_session, top_context, expression, expected)
     )
 
     assert result == expected
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "expression, expected",
+    [
+        (
+            "new RegExp(/foo/)",
+            {
+                "type": "regexp",
+                "value": {
+                    "pattern": "foo",
+                    "flags": "",
+                },
+            },
+        ),
+        (
+            "new Date(1654004849000)",
+            {
+                "type": "date",
+                "value": "2022-05-31T13:47:29.000Z",
+            },
+        ),
+        (
+            "[1, '2', true, new RegExp(/foo/g), [1]]",
```

Comment:
nitpick: to make the test easier to parse, can we use something else than numbers for the strings? (foo, bar, etc...)
Unless we really want to check that "2" doesn't get converted to the number 2.

Reply:
> nitpick: to make the test easier to parse, can we use something else than numbers for the strings? (foo, bar, etc...) Unless we really want to check that "2" doesn't get converted to the number 2.

Agree. But the test "check that string with number value is serialised as a string" should be separate.

Reply:
Alright, I've updated all the complex objects with non-number strings and added the "check that string with number value is serialised as a string" into primitive value test